src/Model/PhasorDynamics/Governor/Tgov1/Tgov1.cpp

```diff
@@ -31,15 +32,22 @@ namespace GridKit
        */
       template <class ScalarT, typename IdxT>
       Tgov1<ScalarT, IdxT>::Tgov1(machine_type* machine, const model_data_type& data)
-        : machine_(machine),
-          R_(data.R),
-          Pvmin_(data.Pvmin),
-          Pvmax_(data.Pvmax),
-          T1_(data.T1),
-          T2_(data.T2),
-          T3_(data.T3),
-          Dt_(data.Dt)
+        : machine_(machine)
       {
+        if (data.parameters.contains(model_data_type::Parameters::R))
+          R_ = std::get<real_type>(data.parameters.at(model_data_type::Parameters::R));
+        if (data.parameters.contains(model_data_type::Parameters::Pvmin))
+          Pvmin_ = std::get<real_type>(data.parameters.at(model_data_type::Parameters::Pvmin));
+        if (data.parameters.contains(model_data_type::Parameters::Pvmax))
+          Pvmax_ = std::get<real_type>(data.parameters.at(model_data_type::Parameters::Pvmax));
+        if (data.parameters.contains(model_data_type::Parameters::T1))
+          T1_ = std::get<real_type>(data.parameters.at(model_data_type::Parameters::T1));
+        if (data.parameters.contains(model_data_type::Parameters::T2))
+          T2_ = std::get<real_type>(data.parameters.at(model_data_type::Parameters::T2));
+        if (data.parameters.contains(model_data_type::Parameters::T3))
+          T3_ = std::get<real_type>(data.parameters.at(model_data_type::Parameters::T3));
+        if (data.parameters.contains(model_data_type::Parameters::Dt))
+          Dt_ = std::get<real_type>(data.parameters.at(model_data_type::Parameters::Dt));
 
```

Comment:
while this is syntactically fine, i think that these `if` statements should have their bodies enclosed by curly braces (`{`, `}`) so that this change is in line with the code style in the rest of the repository (i think most of our code, if not all is this way even for single-line bodies)

Reply:
Same as for the exciter PR, please move parameter assignments to a private method and call that method in the constructor. We want to keep constructors less cluttered.